Author: Pino Toscano <pino@debian.org>
Description: Use -fcf-protection only on x86 architectures
 It does not exist elsewhere.
Forwarded: no
Last-Update: 2020-08-09

--- a/cmake/compilerFlags.cmake
+++ b/cmake/compilerFlags.cmake
@@ -26,7 +26,10 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Li
         # This fails under Fedora, MinGW GCC 8.3.0 and CYGWIN/MSYS 9.3.0
         if (NOT (MINGW OR CMAKE_HOST_SOLARIS OR CYGWIN OR MSYS) )
             if (COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0)
-                add_compile_options(-fstack-clash-protection -fcf-protection)
+                add_compile_options(-fstack-clash-protection)
+                if (CMAKE_SYSTEM_PROCESSOR MATCHES "i686|x86_64")
+                    add_compile_options(-fcf-protection)
+                endif()
             endif()
 
             if( (COMPILER_IS_GCC   AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0) # Not in GCC 4.8
